api/turing/api/alerts_api.go

```diff
@@ -104,7 +112,11 @@ func (c *AlertsController) UpdateAlert(r *http.Request, vars map[string]string,
 	updateAlert := body.(*models.Alert)
 	updateAlert.ID = alert.ID
 	updateAlert.Service = c.getService(*router)
-	if err := c.AlertService.Update(*updateAlert, email); err != nil {
+	dashboardURL, err := c.getDashboardURL(c.AlertService.GetDashboardURLTemplate(), alert, router, nil)
```

Comment:
Do you think we can get the `project`, `environment`, `router` and `routerVersion` in the controller and call `AlertService` to get the dashboard URL directly instead of getting the template from Alert Service and formatting it here?

So in the AlertService interface, we can have something like:
```
    GetDashboardURL(
        alert *models.Alert,
        project *mlp.Project,
        environment *merlin.Environment,
	router *models.Router,
	routerVersion *models.RouterVersion,
    )
```
instead of
```
    GetDashboardURLTemplate() template.Template
```

I think this may be more re-usable if we are populating the Dashboard URL from the API in other responses as an annotation (say, Router / Router Version info will append the dashboard link for the UI to use as well), then this `AlertService.GetDashboardURL` may be called from other controllers with the necessary DB model data, most of which they may already have. wdyt?

Reply:
Btw, will there be another PR for letting the UI fetch the dashboard link from the API so we can remove [this](https://github.com/gojek/turing/blob/main/ui/src/hooks/useMonitoring.js#L13)?

Reply:
OK I've updated AlertService with GetDashboardURL method.

Yes for the UI to be able to get the dashboard URL from the server, we agreed to add the annotations field to Router and RouterVersion object. But this requires db migration and changes in schema. So I think it's better to have it in a separate pull request.